src/gromo/utils/gpu_tracking.py

```diff
@@ -0,0 +1,86 @@
+from types import TracebackType
+from typing import Optional, Self, Type
+
+import numpy as np
+from codecarbon import OfflineEmissionsTracker
+
+from gromo.utils.logger import Logger
+
+
+class GpuTracker:
+    """Tracking GPUs for performance
+
+    Parameters
+    ----------
+    gpu_index : list[int], optional
+        indices of gpus to track, by default [0]
+    interval : int, optional
+        time interval between measuring, by default 15
+    country_iso_code : str, optional
+        the country of the cluster in iso code to estimate carbon emissions
+    logger : Logger | None, optional
+        associated logger to track metrics, by default None
+    """
+
+    def __init__(
+        self,
+        gpu_index: list[int] = [0],
+        interval: int = 15,
+        country_iso_code: str = "FRA",
```

Comment:
I think this should be set in global settings, but this is outside of the scope of this PR.

Could you open an issue,  or add this to the issue for the global settings, to remember that we need to take care of it?

For the global settings, we could inspire from codecarbon code: https://mlco2.github.io/codecarbon/usage.html#configuration

Reply:
Opened at #30 !

Reply:
thanks